perun-ldapc/src/main/java/cz/metacentrum/perun/ldapc/processor/impl/FacilityAttributeProcessor.java

```diff
@@ -37,12 +39,23 @@ public void processAttributeSet(String msg, MessageBeans beans) {
 			log.debug("Getting resources assigned to facility {}", beans.getFacility().getId());
 			// List<Resource> resources = Rpc.FacilitiesManager.getAssignedResources(ldapcManager.getRpcCaller(), beans.getFacility());
 			List<Resource> resources = perun.getFacilitiesManager().getAssignedResources(ldapcManager.getPerunSession(), beans.getFacility());
+
+			Attribute virtAttributeWithValue = null;
+			if (beans.getAttribute().getNamespace().contains(":virt")) {
+				virtAttributeWithValue = ldapcManager.getPerunBl().getAttributesManager().getAttribute(ldapcManager.getPerunSession(),
```

Comment:
You should use Bl layer to prevent unnecessary nested transaction creation. Like this: ``((PerunBl)ldapcManager.getPerunBl()).getAttributesManagerBl().getAttribute(...)``